etc/systemd/system-generators/zfs-mount-generator.c

```diff
@@ -277,21 +277,23 @@ line_worker(char *line, const char *cachefile)
 		if (strcmp(dataset, p_encroot) == 0) {
 			const char *keymountdep = NULL;
 			bool is_prompt = false;
+			bool need_network = false;
 
 			regmatch_t uri_matches[3];
 			if (regexec(&uri_regex, p_keyloc,
 			    nitems(uri_matches), uri_matches, 0) == 0) {
+				p_keyloc[uri_matches[1].rm_eo] = '\0';
 				p_keyloc[uri_matches[2].rm_eo] = '\0';
+				const char *scheme =
+				    &p_keyloc[uri_matches[1].rm_so];
 				const char *path =
 				    &p_keyloc[uri_matches[2].rm_so];
 
-				/*
-				 * Assumes all URI keylocations need
-				 * the mount for their path;
-				 * http://, for example, wouldn't
-				 * (but it'd need network-online.target et al.)
-				 */
-				keymountdep = path;
+				if (strcmp(scheme, "https") == 0 ||
+				    strcmp(scheme, "http") == 0)
+					need_network = true;
```

Comment:
This is fine, but I fear that as more features show up this will get messy.

In the interest of keeping ourselves sane, I'd suggest either:

 - switching over to a collection of `need_X` booleans, and then assembling them when building up the stream; or
 - properly managing the whole string of wants

I think doing both sets a bad precedent.  I'm leaning towards a bunch of booleans, since it avoids a ton of duplicate (and human error prone) string manipulations.

Reply:
It's fine to have multiple Wants & After lines. So what if we just decoupled these (in the code later) by writing a separate Wants and After line for the network-online.target? The code could be simplified by going back to `Wants=%s` unconditionally and letting it be blank if the `wants` variable is empty (as it does now), and likewise for After. That would avoid the need for string manipulation.

Reply:
Indeed! And it works out quite nice, too!